### PR TITLE
Modernize setup.py

### PR DESCRIPTION
**Pull request by dhimmel**
Awesome project.

Here are some changes to make setup.py a bit more modern and readable.

Still have to test locally.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,44 +1,38 @@
...
+    description='Acccounting, asset management and tax report helper for cryptocurrencies',
     license='BSD-3',
     keywords='accounting tax-report portfolio asset-management cryptocurrencies',
     url='https://github.com/rotki/rotki',
     packages=find_packages('.'),
     package_data={
         "rotkehlchen": ["data/*.json"],
     },
-    install_requires=install_requirements,
+    python_requires='>=3.6',
```
Comment (dhimmel) [inline setup.py:30]: didn't know what the minimum version of python this package intends to support? I see CI uses

https://github.com/rotki/rotki/blob/9a3ea165b851414059f271f192b4cb190c8a3c70/.travis.yml#L6-L7

But is that the _only_ supported version or just _one_ supported version?

File: setup.py [hunk 2]
```diff
@@ -1,44 +1,38 @@
...
-
-install_requirements = list(set(
-    install_requires_replacements.get(requirement.strip(), requirement.strip())
-    for requirement in open('requirements.txt')
-    if not requirement.lstrip().startswith('#')
-))
+# Load install_requires from requirements.txt.
+# https://stackoverflow.com/a/59971236/4651668
+requirements = directory.joinpath('requirements.txt').read_text()
+requirements = [str(r) for r in parse_requirements(requirements)]
```
Comment (dhimmel) [inline setup.py:14]: running this on the text from https://github.com/rotki/rotki/blob/9a3ea165b851414059f271f192b4cb190c8a3c70/requirements.txt, I get:

```
['gevent==1.5a2', 'greenlet==0.4.16', 'web3==5.11.1', 'pysqlcipher3==1.0.3', 'requests==2.20.0', 'pycryptodome==3.6.6', 'pycryptodomex==3.6.6', 'coincurve==13.0.0', 'typing-extensions==3.7.4.2', 'base58check==1.0.2', 'bech32==1.2.0', 'gql==2.0.0', 'flask-cors==3.0.8', 'flask-restful==0.3.7', 'flask==1.1.1', 'marshmallow==3.2.1', 'webargs==6.0.0', 'setuptools==41.0.1', 'eth-utils==1.8.4', 'rlp==1.2.0']
```

Also confirmed environment markers, as suggested at https://github.com/rotki/rotki/pull/1246#discussion_r507195675, don't get stripped by parse_requirements.

File: setup.py (hunk 1 above)
Comment (LefterisJP) [reply setup.py:30]: That's a very good question. I can't recall at the moment but we use f-strings, so python 3.6 and up is definitely required. Don't think we got something 3.7 specific yet.

File: setup.py (hunk 2 above)
Comment (LefterisJP) [reply setup.py:14]: Thanks, this actually looks cleaner.

**Review by LefterisJP — APPROVED** [review]
Hey @dhimmel thank you for taking a look. Looks like a good improvement. Thank you for the patch.

I normally would be a bit careful in merging something in setup.py but we will test it soon in a new release and it looks okay from a quick look.

**Pull request merged**